controllers/storagecluster/storagecluster_controller.go

```diff
@@ -144,28 +148,91 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+			}
+
+			for _, cond := range oldObj.Status.Conditions {
+				cond.LastHeartbeatTime = metav1.Time{}
+				cond.LastTransitionTime = metav1.Time{}
+			}
+
+			oldObj.ObjectMeta.ResourceVersion = ""
+			newObj.ObjectMeta.ResourceVersion = ""
+
+			return !reflect.DeepEqual(oldObj, newObj)
+		},
+	}
+
+	build := ctrl.NewControllerManagedBy(mgr).
 		For(&ocsv1.StorageCluster{}, builder.WithPredicates(scPredicate)).
-		Owns(&cephv1.CephCluster{}).
-		Owns(&corev1.PersistentVolumeClaim{}, builder.WithPredicates(pvcPredicate)).
+		Owns(&cephv1.CephCluster{}, builder.WithPredicates(cephClusterIgnoreTimeUpdatePredicate)).
+		Owns(&cephv1.CephBlockPool{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephFilesystem{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephFilesystemSubVolumeGroup{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephNFS{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephObjectStore{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephObjectStoreUser{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephRBDMirror{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.PersistentVolumeClaim{}, builder.WithPredicates(predicate.GenerationChangedPredicate{}, pvcPredicate)).
 		Owns(&appsv1.Deployment{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Owns(&corev1.Service{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Owns(&corev1.ConfigMap{}, builder.MatchEveryOwner, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Owns(&corev1.Secret{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
-		Watches(&ocsclientv1a1.StorageClient{}, enqueueStorageClusterRequest).
-		Watches(&ocsv1.StorageProfile{}, enqueueStorageClusterRequest).
+		Owns(&routev1.Route{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&templatev1.Template{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&storagev1.StorageClass{}, enqueueStorageClusterRequest, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&volumesnapshotv1.VolumeSnapshotClass{}, enqueueStorageClusterRequest, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&ocsclientv1a1.StorageClient{}, enqueueStorageClusterRequest, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&ocsv1.StorageProfile{}, enqueueStorageClusterRequest, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Watches(
 			&extv1.CustomResourceDefinition{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "virtualmachines.kubevirt.io",
 				},
 			},
 			enqueueStorageClusterRequest,
+			builder.WithPredicates(predicate.GenerationChangedPredicate{}),
```

Comment:
@iamniting I am not sure generation will change for metadata changes, like labels and annotation (I am quite sure the opposite is correct)

Reply:
removed it.